libnmstate/nm/applier.py

```diff
@@ -79,28 +81,33 @@ def prepare_edited_ifaces_configuration(ifaces_desired_state):
 
 
 def set_ifaces_admin_state(ifaces_desired_state):
+    devs_actions = {}
     for iface_desired_state in ifaces_desired_state:
+        devs = _get_devs_from_iface_state(iface_desired_state)
         if iface_desired_state['state'] == 'up':
-            _set_dev_state(iface_desired_state, device.activate)
+            devs_actions.update({dev: device.activate for dev in devs})
         elif iface_desired_state['state'] == 'down':
```

Comment:
Maybe check `iface_desired_state['state'] in ('down', 'absent')` to show that the same action happens for both cases.

Reply:
Done